test/src/test/java/org/corfudb/integration/WorkflowIT.java

```diff
@@ -141,4 +155,127 @@ public void AddAndRemoveNodeIT() throws Exception {
         shutdownCorfuServer(server_2);
         shutdownCorfuServer(server_3);
     }
+
+    //@Test
+    public void clusterResizingTest1() throws Exception {
+        // This tests will resize the cluster according to the following
+        // order: add node -> add node -> remove node -> add node -> add node
+        final int n0Port = 9000;
+        final int n1Port = 9001;
+        final int n2Port = 9002;
+
+        final int clusterSizeN1 = 1;
+        final int clusterSizeN2 = 2;
+        final int clusterSizeN3 = 3;
+
+        new CorfuServerRunner()
+                .setHost(host)
+                .setPort(n0Port)
+                .setSingle(true)
+                .runServer();
+
+        new CorfuServerRunner()
+                .setHost(host)
+                .setPort(n1Port)
+                .setSingle(false)
+                .runServer();
+
+        new CorfuServerRunner()
+                .setHost(host)
+                .setPort(n2Port)
+                .setSingle(false)
+                .runServer();
+
+        CorfuRuntime n0Rt = new CorfuRuntime(getConnectionString(n0Port)).connect();
+        CorfuTable table = n0Rt.getObjectsView().build()
+                .setType(CorfuTable.class)
+                .setStreamName("table1").open();
+
+        final int iter = 1000;
+        for (int x = 0; x < iter; x++) {
+            table.put(String.valueOf(x), String.valueOf(x));
+        }
+
+        n0Rt.getManagementView().addNode(getConnectionString(n1Port), workflowNumRetry,
+                timeout, pollPeriod);
+        n0Rt.invalidateLayout();
+
+        assertThat(n0Rt.getLayoutView().getLayout().getAllServers().size()).isEqualTo(clusterSizeN2);
+
+        n0Rt.getManagementView().forceRemoveNode(getConnectionString(n1Port), workflowNumRetry,
+                timeout, pollPeriod);
+        n0Rt.invalidateLayout();
+
+        assertThat(n0Rt.getLayoutView().getLayout().getAllServers().size()).isEqualTo(clusterSizeN1);
+
+        n0Rt.getManagementView().addNode(getConnectionString(n1Port), workflowNumRetry,
+                timeout, pollPeriod);
+
+        n0Rt.getManagementView().addNode(getConnectionString(n2Port), workflowNumRetry,
+                timeout, pollPeriod);
+
+        n0Rt.invalidateLayout();
+        assertThat(n0Rt.getLayoutView().getLayout().getAllServers().size()).isEqualTo(clusterSizeN3);
+
+        n0Rt.getManagementView().removeNode(getConnectionString(n1Port), workflowNumRetry,
+                timeout, pollPeriod);
+        n0Rt.invalidateLayout();
+        assertThat(n0Rt.getLayoutView().getLayout().getAllServers().size()).isEqualTo(clusterSizeN2);
+
+        for (int x = 0; x < iter; x++) {
+            assertThat(table.get(String.valueOf(x))).isEqualTo(String.valueOf(x));
+        }
+    }
+
+    //@Test
```

Comment:
These are commented out and have no comments.

Also we need to have tests for hand of god case.  3 node cluster, we have 2 node failures and now we need to shrink the cluster down to the last standing node.

Reply:
I have a test that does that, look at clusterResizingTest2.